docs/examples/10_inverse_rendering/invert_caustic.py

```diff
@@ -193,18 +198,88 @@ def load_ref_image(config, resolution, output_dir):
 
     image_ref = np.array(b)
 
-    write_bitmap(os.path.join(output_dir, 'out_ref.exr'), image_ref, resolution)
+    write_bitmap(join(output_dir, 'out_ref.exr'), image_ref, resolution)
     return mitsuba.core.Float(image_ref.ravel())
 
-def main(config):
+
+def create_flat_lens_mesh(resolution):
+    from mitsuba.core import Float, UInt32, ScalarTransform4f
+    from mitsuba.render import Mesh
+
+    assert resolution[0] == resolution[1], 'Not supported yet: non-symmetric lens mesh resolution'
+    X, Y = np.mgrid[:resolution[0], :resolution[1]]
+    # Scale UVs to [-1, 1]
+    U = X / (resolution[0]-1)
+    V = Y / (resolution[1]-1)
+    # Scale actual positions to [-1, 1]
+    X = 2 * (U - 0.5)
+    Y = 2 * (V - 0.5)
+
+    n_vertices = resolution[0] * resolution[1]
+
+    vertices = np.stack([
+        X.ravel(), Y.ravel(), np.zeros((n_vertices))
+    ], axis=1)
+    # # Rotate to match the `slab.obj` positioning
+    # to_world = ScalarTransform4f.rotate(axis=(1, 0, 0), angle=90)
+    # to_world = np.array(to_world.matrix)[:3, :3]
+    # vertices = (to_world @ vertices.T).T
+    assert vertices.shape == (n_vertices, 3)
+
+    texcoords =  np.stack([
+        U.ravel(), V.ravel()
+    ], axis=1)
+    assert texcoords.shape == (n_vertices, 2)
+
```

Comment:
Would it be possible to write this using `enoki` instead of `numpy`? It would be a great example on how to use the enoki Python API :)